CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ option(FINUFFT_ENABLE_SANITIZERS "Whether to enable sanitizers, only effective f
 option(FINUFFT_USE_OPENMP "Whether to use OpenMP for parallelization. If disabled, the finufft library will be single threaded. This does not affect the choice of FFTW library." ON)
 option(FINUFFT_USE_CUDA "Whether to build CUDA accelerated FINUFFT library (libcufinufft). This is completely independent of the main FINUFFT library" OFF)
 option(FINUFFT_USE_CPU "Whether to build the ordinary FINUFFT library (libfinufft)." ON)
+option(FINUFFT_STATIC_LINKING "Whether to link the static FINUFFT library (libfinufft_static)." OFF)
```

Comment:
Yes, for users they should use whatever they want to link with and just default to static link.
Here adding this `FINUFFT_STATIC_LINKING` just to make a choice for our tests/examples code to link dlls and by default I just want to make the harder case(generate dll and link) CI passes, it will be good to see unexpected behavior/error in the ci runs instead of users see the unexpected behavior and notify us.
We can make static default here and change it in ci *.yml scripts to test the dynamic link.